### PR TITLE
fix cascading issue

### LINKED ISSUE #178: [PE-D][Tester E] Meeting still appears on "Meetings" list when associated person is deleted
When a meeting is added for a person, it appears on the "meetings" list.

However, when the person is deleted, the associated meeting still appears on the "meetings" list, until the app restarts. However, the associated meeting is still in the json file, which may cause unexpected behaviours in the future.

To replicate this:

1. This is the default address book with the initial data.

2. Enter this command: `addMeeting clientIndex/1 dt/02-01-2025 12:00 d/Sign life plan` to add a meeting to "Alex Yeoh", and the meeting should appear on the meeting list.

3. However, if this command is entered: `delete 1` to delete "Alex Yeoh" from the person list, the associated meeting is still on the "meeting list".

4. After further investigation, it seems like the meeting is not being removed from the "meetings" field in the saved json file, when it should be removed.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: tohlh/ped#2

### LINKED ISSUE #217: [PE-D][Tester B] Deletion of person does not cascade down to meetings associated with person
I have deleted Ibrahim, yet his meeting is still displayed, I think this is a feature flaw.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: Ko-Khan/ped#9

### PR DESCRIPTION
**Pull request by jaejayrome**
closes #217 
fixes #178

### CODE REVIEW

**Review by sethteo — COMMENTED** [review]
LGTM!

**Review by chewbum — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deleteAllMeetingsForClient(targetIndex);
```
Comment (timothysashimi) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:45]: good idea of cascading the deletion of meetings when a client is deleted!

**Comment by timothysashimi** [comment]
LGTM!

**Pull request merged**